src/cljs/harja/views/urakka/kulut.cljs

```diff
@@ -542,26 +554,44 @@
       [:div.palsta
        [:h3 {:style {:width "100%"}}
         "Mihin työhön kulu liittyy?"]
-       [:div.flex-row
-        [:input#kulu-normaali.vayla-radio
-         {:type            :radio
-          :name            "kulu-group"
-          :default-checked (cond
-                             (> (count kohdistukset) 1) false
-                             (= "lisatyo" (:maksueratyyppi (first kohdistukset))) false
-                             :else true)
-          :disabled        (not= 0 haetaan)
-          :on-change       #(let [kohdistusten-paivitys-fn (if (.. % -target -checked)
-                                                             resetoi-kohdistukset)
-                                  jalkiprosessointi-fn (if (.. % -target -checked)
-                                                         (fn [lomake]
-                                                           (vary-meta
-                                                             lomake
-                                                             maaramitallisen-validoinnit
-                                                             {:lomake  lomake
-                                                              :indeksi 0})))]
-                              (paivitys-fn {:jalkiprosessointi-fn jalkiprosessointi-fn} :kohdistukset kohdistusten-paivitys-fn))}]
-        [:label {:for "kulu-normaali"} "Normaali suunniteltu tai määrämitattava hankintakulu"]]
+       [vayla-radio {:id              "kulu-normaali"
+                     :teksti          "Normaali suunniteltu tai määrämitattava hankintakulu"
+                     :ryhma           "kulu-group"
+                     :oletus-valittu? (cond
+                                        (> (count kohdistukset) 1) false
+                                        (= "lisatyo" (:maksueratyyppi (first kohdistukset))) false
+                                        :else true)
+                     :disabloitu?     (not= 0 haetaan)
+                     :muutos-fn       #(let [kohdistusten-paivitys-fn (if (.. % -target -checked)
```

Comment:
Jos määrittää anonyymin funktion näin argumenttina jollekin Reagentin komponentille, niin tuo komponentti (`vayla-radio`) renderöidään joka kerta, kun tämä (`tehtavien-syotto`) komponentti renderöidään vaikkei argumentit olisikaan muuttunut. Tää `muutos-fn` kannattaisi siis määrittää joko tämän (`tehtavien-syotto`) renderöinti funktion ulkopuolella tai sitten wrapata se `reagent.core/partial`:in sisään.

Reply:
hyvä muistutus, nämä fiksailtu